app/models/credit_card.rb

```diff
@@ -1,12 +1,14 @@
 class CreditCard < ApplicationRecord
+  before_save :set_last_four
 
   attr_accessor :number
 
-  before_create :set_last_four
   validates :first_name, :last_name, :month, :year, presence: true
   belongs_to :user
 
+  private
+
   def set_last_four
-    self.last_four = number[-4..-1]
```

Comment:
when assigning you do need to do `this.`, otherwise it just creates a local variable